l10n_it_ricevute_bancarie/models/account/account.py

```diff
@@ -0,0 +1,323 @@
+# Copyright (C) 2012 Andrea Cometa.
+# Email: info@andreacometa.it
+# Web site: http://www.andreacometa.it
+# Copyright (C) 2012 Associazione OpenERP Italia
+# (<http://www.odoo-italia.org>).
+# Copyright (C) 2012-2018 Lorenzo Battistini - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, _
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import Warning as UserError
+
+
+class AccountPaymentTerm(models.Model):
+    # flag riba utile a distinguere la modalità di pagamento
+    _inherit = 'account.payment.term'
+
+    riba = fields.Boolean('Riba', default=False)
+    riba_payment_cost = fields.Float(
+        'RiBa Payment Cost', digits=dp.get_precision('Account'),
+        help="Collection fees amount. If different from 0, "
+             "for each payment deadline an invoice line will be added "
+             "to invoice, with this amount"
+    )
+
+
+class ResBankAddField(models.Model):
+    _inherit = 'res.bank'
+    banca_estera = fields.Boolean('Banca Estera')
+
+
+class ResPartnerBankAdd(models.Model):
+    _inherit = 'res.partner.bank'
+    codice_sia = fields.Char(
+        "SIA Code", size=5,
+        help="Identification Code of the Company in the System Interbank")
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+    riba_accredited_ids = fields.One2many(
+        'riba.distinta', 'accreditation_move_id', 'Distinte RiBa accredited',
+        readonly=True)
+    riba_unsolved_ids = fields.One2many(
+        'riba.distinta.line', 'unsolved_move_id', 'Distinte RiBa unsolved',
+        readonly=True)
+
+
+# se distinta_line_ids == None allora non è stata emessa
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    distinta_line_ids = fields.One2many(
+        'riba.distinta.move.line', 'move_line_id', "Dettaglio riba")
+    riba = fields.Boolean(
+        related='invoice_id.payment_term_id.riba', string='RiBa', store=False)
+    unsolved_invoice_ids = fields.Many2many(
+        'account.invoice', 'invoice_unsolved_line_rel', 'line_id',
+        'invoice_id', 'Unsolved Invoices')
+    iban = fields.Char(
+        related='partner_id.bank_ids.acc_number', string='IBAN', store=False)
```

Comment:
Non so se questo PR e ancora attivo. In ogni caso, il cliente può avere vari conti bancari, se in uno ha un mandato anullato, il related non lo sa e mete un iban non valido.

Reply:
> Non so se questo PR e ancora attivo. In ogni caso, il cliente può avere vari conti bancari, se in uno ha un mandato anullato, il related non lo sa e mete un iban non valido.

Grazie. Sai se il problema è anche sulla 12?

Reply:
Forse no, il iban si ottiene di un altro modo, ma sembra anche di ottenere il primo conto che trova senza vedere il mandato. Ma come ho detto non sono sicuro, non ho una 12 per provare adesso.